src/transformers/deepspeed.py

```diff
@@ -36,7 +34,13 @@ def is_deepspeed_available():
     return importlib.util.find_spec("deepspeed") is not None
 
 
-class HfDeepSpeedConfig:
+if is_accelerate_available() and is_deepspeed_available():
+    from accelerate.utils.deepspeed import HfDeepSpeedConfig as DeepSpeedConfig
+else:
+    from builtins import object as DeepSpeedConfig
```

Comment:
Apologies if this has already been discussed, shouldn't it assert if the conditions on line 37 fail?

`HfDeepSpeedConfig` inheriting from `object` will be a bogus class.

Reply:
Hello @stas, 

The thing is that this for any call of `from_pretrained`, it calls `from .deepspeed import is_deepspeed_zero_stage3`. Now, this call will try import from accelerate and as such fail. If we assert, it will fail again for all calls of `from_pretrained` or `from_config` even if we don't use deepspeed.

And having `accelerate` as hard dependency was causing a lot of Tensorflow tests to fail because installing accelerate will install torch and hence torch tests were being incorrectly run for Tensorflow tests.

When `deepspeed` is being used, assert check is made in `training_args.py` as well in `transformers.deepspeed.HFDeepSpeedConfig` init functions. 

So, when using deepspeed necessary checks are done and everything works as expected.

When not using deepspeed, it is fine if HFDeepSpeedConfig inherits `object` as it will never be used.

Reply:
I thought that there must have been a non-obvious reason for this - thank you for explaining that in details, @pacman100 

Would you kindly document that on line 40, as for a reader that code is bogus w/o the notes. Basically, something like:

```
# will inherit from a dummy `object` if accelerate is not available, so that python succeeds import this file
# deepspeed glue code will not ever use that dummy object as it checks if accelerate is available
```

Much appreciated.